sdk/pinocchio/src/sysvars/slot_hashes.rs

```diff
@@ -0,0 +1,173 @@
+pub const PAIR_SIZE: usize = SLOT_SIZE + HASH_SIZE;
+
+/// Size of the vector length field in bytes.
+pub const VEC_LENGTH_SIZE: usize = core::mem::size_of::<u64>();
+
+#[repr(C)]
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub struct SlotHashEntry {
+    /// slot number
+    pub slot: Slot,
+
+    /// hash value
+    pub hash: Hash
+}
+
+/// A struct to access the slot hashes sysvar.
+///
+/// This struct provides efficient methods to access specific entries in the
+/// slot hashes sysvar without deserializing the entire data structure, which
+/// would be too large to process on-chain.
+#[derive(Debug, Default)]
+pub struct SlotHashesSysvar;
+
+impl SlotHashesSysvar {
+    /// Returns the number of entries in the slot hashes sysvar.
+    ///
+    /// Returns `None` if the sysvar syscall fails.
+    pub fn len() -> Option<usize> {
+      let mut len_buf = [0u8; VEC_LENGTH_SIZE];
+
+      let result = unsafe {
+          crate::syscalls::sol_get_sysvar(
+            SLOT_HASHES_ID.as_ptr(),
+            len_buf.as_mut_ptr(),
+            0,
+            VEC_LENGTH_SIZE as u64,
+          )
+      };
+
+      if result != 0 {
+        return None;
+      }
+
+      let len = u64::from_le_bytes(len_buf);
+      Some(core::cmp::min(len as usize, MAX_ENTRIES))
+    }
+
+    /// Checks if the slot hashes sysvar is empty.
+    ///
+    /// Returns `None` if the sysvar syscall fails.
+    pub fn is_empty()-> Option<bool> {
+        Self::len().map(|len| len == 0)
+    }
+
+    /// Gets the slot hash entry at the specified index.
+    ///
+    /// Returns `None` if the index is out of bounds or the sysvar syscall fails.
+    pub fn get_entry(index: usize) -> Option<SlotHashEntry> {
+        let len = Self::len()?;
+
+        if index >= len {
+            return None;
+        }
+
+        let offset = (VEC_LENGTH_SIZE + (index * PAIR_SIZE)) as u64;
+        let mut entry_buf = [0u8, PAIR_SIZE as u8];
+
+        let result = unsafe {
+            crate::syscalls::sol_get_sysvar(
+                SLOT_HASHES_ID.as_ptr(), 
+                entry_buf.as_mut_ptr(), 
+                offset, 
+                PAIR_SIZE as u64
+            )
+        };
+
+        if result != 0 {
+            return None;
+        }
+
+        // First 8 bytes are the Slot
+        let mut slot_bytes = [0u8; SLOT_SIZE];
+        slot_bytes.copy_from_slice(&entry_buf);
+        let slot = u64::from_le_bytes(slot_bytes);
+
+        // Next 32 bytes are the hash
+        let mut hash = [0u8; HASH_SIZE];
+        hash.copy_from_slice(&entry_buf[SLOT_SIZE..]);
+
+        Some(SlotHashEntry { slot, hash })
+    }
+
+    /// Gets the hash for the specified slot.
+    ///
+    /// This performs a linear search through the entries.
+    /// Returns `None` if the slot is not found or the sysvar syscall fails.
+    pub fn get_hash(target_slot: Slot) -> Option<Hash> {
+        let len = Self::len()?;
+
+        for i in 0..len {
+            if let Some(entry) = Self::get_entry(i) {
```

Comment:
This approach works, but it will be expensive to use multiple syscalls to iterate over them. It will be better to follow a similar approach as [PodSlotHashes](https://github.com/anza-xyz/solana-sdk/blob/0666fa5999750153070e5c43d64813467bfdc38e/sysvar/src/slot_hashes.rs#L106-L145) and [Instructions](https://github.com/anza-xyz/pinocchio/blob/main/sdk/pinocchio/src/sysvars/instructions.rs) sysvar.